### PR TITLE
[REVIEW] Add column feature hashing support

### PR DESCRIPTION
**Pull request by cmgreen210**
Closes https://github.com/rapidsai/cudf/issues/798

- [x] Update `gdf_hash` with initial hash values parameter.
- [x]  Add python API for for hashing column values with initial hash value set to hashed col name.

### CODE REVIEW

**Comment by cmgreen210** [comment]
@kkraus14 @nsakharnykh @muellren this is my start.

**Comment by kkraus14** [comment]
add to whitelist

File: cpp/src/hash/hashing.cu [hunk 1]
```diff
@@ -150,6 +163,65 @@ gdf_error gdf_hash(int num_cols, gdf_column **input, gdf_hash_func hash, gdf_col
...
+  hash_value_type * p_output = static_cast<hash_value_type*>(output->data);
+  thrust::device_ptr<hash_value_type> col_hash_values = thrust::device_pointer_cast(p_output);
+
+  hash_value_type init_val{0};
+  switch(hash)
+  {
+    case GDF_HASH_MURMUR3:
+      {
+        if (hash_col_name) {
+          init_val = MurmurHash3_32<char *>{}(input->col_name);
```
[the remark this answers is not in the document]
Comment (cmgreen210) [reply cpp/src/hash/hashing.cu]: it is not. the python column representation doesn't set the `col_name` field anyways so I'll be removing this in favor of setting the initial hash values outside of the function call.

**Comment by cmgreen210** [comment]
@jrhemstad that's a good point. Curious why you think it's better to leave the modulo operation outside of this call.

**Comment by jrhemstad** [comment]
> @jrhemstad that's a good point. Curious why you think it's better to leave the modulo operation outside of this call.

It just adds extra complexity to the API and implementation. I'm resistant to specializing a function for a specific use-case when it can be easily done via the composition of other primitive functions. 

That said, if it turns out that doing the modulo in a second pass has a non-negligible performance impact, then we can explore adding the modulo into `gdf_hash`. I'd just to prefer to err on the side of a simpler API and implementation until profiling tells us that we need to something else.

**Comment by cmgreen210** [comment]
@jrhemstad sounds good! I'll start making the changes.

**Comment by harrism** [comment]
@cmgreen210 why all the force pushes?

**Comment by cmgreen210** [comment]
@harrism This is my first time working through github pull requests. Guess I'm not familiar with the correct workflow. Sorry if this caused any issues. How should I handle this going forward?

**Comment by harrism** [comment]
Push without forcing unless you absolutely need to. Ever now and then you should merge the branch (in this case `branch-0.6`) from upstream into your local branch and resolve conflicts locally before pushing.

**Comment by cmgreen210** [comment]
@harrism @jrhemstad closing this to untangle the commit mess I made. Please see https://github.com/rapidsai/cudf/pull/858 instead.

**Pull request closed without merging**